client/FxAccountClient.js

```diff
@@ -1031,4 +1099,30 @@ define([
   return FxAccountClient;
 });
 
+var METRICS_CONTEXT_OPTIONS = {
+  context: true,
+  entrypoint: true,
+  migration: true,
+  service: true,
+  utmCampaign: true,
+  utmContent: true,
+  utmMedium: true,
+  utmSource: true,
+  utmTerm: true
+};
+
+function marshallMetricsContext (data) {
+  var metricsContext = {
+    flowId: data.flowId,
+    flowBeginTime: data.flowBeginTime
+  };
+
+  for (var key in data) {
+    if (data.hasOwnProperty(key) && data[key] && METRICS_CONTEXT_OPTIONS[key]) {
```

Comment:
nit: possible to use `.prototype.hasOwnProperty` here instead? 

Ref: https://github.com/mozilla/fxa-content-server/blob/7d8c2c0661c4492895d39cc10ebcbd602f4bf596/app/scripts/lib/session.js#L46